src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrValidator.java

```diff
@@ -0,0 +1,105 @@
+import uk.gov.hmcts.reform.bulkscanprocessor.model.blob.InputEnvelope;
+import uk.gov.hmcts.reform.bulkscanprocessor.model.blob.InputScannableItem;
+import uk.gov.hmcts.reform.bulkscanprocessor.ocrvalidation.client.OcrValidationClient;
+import uk.gov.hmcts.reform.bulkscanprocessor.ocrvalidation.client.model.req.FormData;
+import uk.gov.hmcts.reform.bulkscanprocessor.ocrvalidation.client.model.req.OcrDataField;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
+
+@Component
+@EnableConfigurationProperties(ContainerMappings.class)
+public class OcrValidator {
+
+    private static final Logger log = LoggerFactory.getLogger(OcrValidator.class);
+
+    private final OcrValidationClient client;
+    private final ContainerMappings containerMappings;
+    private final AuthTokenGenerator authTokenGenerator;
+
+    //region constructor
+    public OcrValidator(
+        OcrValidationClient client,
+        ContainerMappings containerMappings,
+        AuthTokenGenerator authTokenGenerator
+    ) {
+        this.client = client;
+        this.containerMappings = containerMappings;
+        this.authTokenGenerator = authTokenGenerator;
+    }
+    //endregion
+
+    public void assertIsValid(InputEnvelope envelope) {
+        Optionals.ifAllPresent(
+            findValidationUrl(envelope),
+            findDocWithOcr(envelope),
+            (validationUrl, docWithOcr) ->
+                Try.of(() -> client.validate(validationUrl, toFormData(docWithOcr), authTokenGenerator.generate()))
+                    .onSuccess(res -> {
+                        switch (res.status) {
+                            case ERRORS:
+                                throw new OcrValidationException(Strings.join(res.errors, ", "));
+                            case WARNINGS:
+                                log.info("Validation ended with warnings. File name: {}", envelope.zipFileName);
+                                break;
+                            default:
+                                break;
+                        }
+                    })
+                    .onFailure(exc -> {
+                        log.error("Error calling validation endpoint. Url: {}", validationUrl, exc);
+                        // log error and proceed
+                    })
+        );
+    }
+
+    private Optional<String> findValidationUrl(InputEnvelope envelope) {
+        return containerMappings
+            .getMappings()
+            .stream()
+            .filter(mapping -> Objects.equals(mapping.getPoBox(), envelope.poBox))
+            .findFirst()
+            .map(mapping -> mapping.getOcrValidationUrl())
+            .filter(url -> !Strings.isNullOrEmpty(url));
+    }
+
+    private Optional<InputScannableItem> findDocWithOcr(InputEnvelope envelope) {
+        List<InputScannableItem> docsWithOcr =
+            envelope
```

Comment:
just wondering as seen in many places: why is this in new line?

Reply:
why not? ;)
To have `.scannableItems` and all the further operations close to envelope